lib/common/tests/nodes/pcmk_foreach_active_resource_test.c

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2024 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU General Public License version 2
+ * or later (GPLv2+) WITHOUT ANY WARRANTY.
+ */
+
+#include <crm_internal.h>
+
+#include <stdio.h>      // NULL
+#include <glib.h>       // GList, TRUE, FALSE
+
+#include <crm/common/nodes.h>
+#include <crm/common/resources.h>
+#include <crm/common/unittest_internal.h>
+
+static int counter = 1;
```

Comment:
`counter` isn't necessary if you use `expect_function_call()` in the caller and `function_called()` in `fn()`. Both are in cmocka. Up to you. It's slightly clearer but doesn't really matter since nothing else updates `counter`.

Reply:
I did forget about those, but I also use counter to assert the resource IDs are as expected